guacamole-common/src/main/java/org/apache/guacamole/protocol/GuacamoleProtocolVersion.java

```diff
@@ -0,0 +1,193 @@
+     * detection, arbitrary handshake instruction order, and support
+     * for passing the client timezone to the server during the handshake.
+     */
+    VERSION_1_1_0(1, 1, 0);
+    
+    /**
+     * The major version component of the protocol version.
+     */
+    private final int major;
+
+    /**
+     * The minor version component of the protocol version.
+     */
+    private final int minor;
+
+    /**
+     * The patch version component of the protocol version.
+     */
+    private final int patch;
+    
+    /**
+     * Generate a new GuacamoleProtocolVersion object with the given
+     * major version, minor version, and patch version.
+     * 
+     * @param major
+     *     The integer representation of the major version component.
+     * 
+     * @param minor
+     *     The integer representation of the minor version component.
+     * 
+     * @param patch 
+     *     The integer representation of the patch version component.
+     */
+    GuacamoleProtocolVersion(int major, int minor, int patch) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+    }
+    
+    /**
+     * Return the major version component of the protocol version.
+     * 
+     * @return 
+     *     The integer major version component.
+     */
+    public int getMajor() {
+        return major;
+    }
+    
+    /**
+     * Return the minor version component of the protocol version.
+     * 
+     * @return 
+     *     The integer minor version component.
+     */
+    public int getMinor() {
+        return minor;
+    }
+    
+    /**
+     * Return the patch version component of the protocol version.
+     * 
+     * @return 
+     *     The integer patch version component.
+     */
+    public int getPatch() {
+        return patch;
+    }
+    
+    /**
+     * Determines whether or not this object is greater than or equal to the
+     * the version passed in to the method.  Returns a boolean true if the
+     * version is the same as or greater than the other version, otherwise
+     * false.
+     * 
+     * @param otherVersion
+     *     The version to which this object should be compared.
+     * 
+     * @return 
+     *     True if this object is greater than or equal to the other version.
+     */
+    public boolean atLeast(GuacamoleProtocolVersion otherVersion) {
+        
+        // Major version is greater
+        if (major > otherVersion.getMajor())
+            return true;
+        
+        // Major version is less than or equal to.
+        else {
+            
+            // Major version is less than
+            if (major < otherVersion.getMajor())
+                return false;
+            
+            // Major version is equal, minor version is greater
+            if (minor > otherVersion.getMinor())
+                return true;
+            
+            // Minor version is less than or equal to.
+            else {
```

Comment:
Same here - it may be possible to reduce the complexity of this through relying on the way you've already structured the comparisons.

Reply:
Come to think of it, maybe there's yet a more readable, simpler way to achieve this? Just occurred to me (pardon the pseudocode):

    // Compare major with priority
    if (major different)
        return major > otherVersion.major

    // Major is identical, only minor and patch matter
    if (minor different)
        return minor > otherVersion.minor

    // Major and minor are identical, only patch matters
    return patch >= otherVersion.patch

... unless that's wrong. I might be right, or I might need more coffee, or I might be right **and** need more coffee.

Reply:
Well, more coffee never hurt anything.

Ran through several version comparisons, and this seems sane, so I'm going with it.  I knew there had to be a simpler way to do this - thanks!!

Reply:
Implemented based on your pseudocode.